src/common/stream.cc

```diff
@@ -13,16 +13,28 @@
 
 FileBuffer::FileBuffer(const char* fileName)
     : fileName_(fileName)
+    , fd_(-1)
 {
     init(fileName);
 }
 
 FileBuffer::FileBuffer(const std::string& fileName)
     : fileName_(fileName)
+    , fd_(-1)
 {
     init(fileName.c_str());
 }
 
+FileBuffer::~FileBuffer()
+{
+    if( -1 != fd_ ) {
+        int res = ::close(fd_);
+        if (res == -1) {
+            throw std::runtime_error("Could not get file stats");
```

Comment:
I would avoid throwing in a destructor, as throwing during stack unwinding will call end up calling `std::terminate`, which we do not want.